ci/test-conformance-eks.sh

```diff
@@ -0,0 +1,235 @@
+    export KUBECONFIG="${KUBECONFIG_PATH}/kubeconfig"
+
+    sleep 5
+    if [[ $(kubectl get svc) ]]; then
+        echo "=== EKS cluster setup succeeded ==="
+    else
+        echo "=== EKS kubectl is not configured correctly! ==="
+        exit 1
+    fi
+}
+
+function deliver_antrea_to_eks() {
+
+    echo "=== Configuring Antrea for cluster ==="
+
+    # EKS service CIDR is currently assigned by AWS,
+    # either from 172.20.0.0/16 or 10.100.0.0/16 depending on the range of VPC
+    # https://forums.aws.amazon.com/thread.jspa?messageID=859958
+    k8s_svc_addr=$(kubectl get svc -A | grep kubernetes | awk '{print $4}')
+
+    if [[ $k8s_svc_addr == 10.100.* ]]; then
+    	 k8s_svc_cidr='10.100.0.0/16'
+    elif [[ $k8s_svc_addr == 172.20.* ]]; then
+    	 k8s_svc_cidr='172.20.0.0/16'
+    else
+    	 echo "Cannot determine EKS serviceCIDR!"
+         exit 1
+    fi
+
+    set +e
+    worker_node_ip_1=$(kubectl get nodes -o wide | sed -n '2 p' | awk '{print $7}')
+    node_mtu=$(ssh -o StrictHostKeyChecking=no -l "ec2-user" $worker_node_ip_1 \
+      'export PATH=$PATH:/usr/sbin; ip a | grep -E eth0.*mtu | cut -d " " -f5')
+
+    sed -i "s|#defaultMTU: 1450|defaultMTU: ${node_mtu}|g" ../build/yamls/antrea-eks.yml
+    sed -i "s|#serviceCIDR: 10.96.0.0/12|serviceCIDR: ${k8s_svc_cidr}|g" ../build/yamls/antrea-eks.yml
+    echo "defaultMTU set as ${node_mtu}"
+    echo "seviceCIDR set as ${k8s_svc_cidr}"
+
+    kubectl apply -f ../build/yamls/antrea-eks.yml
+    set -e
+
+    kubectl rollout status --timeout=2m deployment.apps/antrea-controller -n kube-system
+    kubectl rollout status --timeout=2m daemonset/antrea-agent -n kube-system
+
+    echo "=== Antrea has been deployed for EKS cluster ${CLUSTER} ==="
+}
+
+function run_conformance() {
+    echo "=== Running Antrea Conformance and Network Policy Tests ==="
+
+    ./run-k8s-e2e-tests.sh --e2e-conformance --e2e-network-policy > eks-test.log
+
+    if grep -Fxq "Failed tests:" eks-test.log
+    then
+        echo "Failed cases exist."
+        TEST_FAILURE=true
+    else
+        echo "All tests passed."
+    fi
+
+    echo "=== Cleanup Antrea Installation ==="
+    for antrea_yml in ../build/yamls/*.yml
+    do
+        kubectl delete -f ${antrea_yml} --ignore-not-found=true || true
+    done
+
+    if [[ "$TEST_FAILURE" == false ]]; then
+        echo "=== SUCCESS !!! ==="
+    fi
+    echo "=== FAILURE !!! ==="
+}
+
+function cleanup_cluster() {
+    echo '=== Cleaning up EKS cluster ${cluster} ==='
+    retry=5
+    while [[ "${retry}" -gt 0 ]]; do
+       eksctl delete cluster --name ${CLUSTER} --region $REGION
+       if [[ $? -eq 0 ]]; then
+         break
+       fi
+       sleep 10
+       retry=$((retry-1))
+    done
+    if [[ "${retry}" -eq 0 ]]; then
+       echo "=== Failed to delete EKS cluster ${CLUSTER}! ==="
+       exit 1
+    fi
+    rm ${KUBECONFIG_PATH}/kubeconfig
+    echo "=== Cleanup cluster ${CLUSTER} succeeded ==="
+}
+
+source ~/.bashrc
```

Comment:
I didn't see this before. What is it for? There may not be a `~/.bashrc` file on the system.

Reply:
I'm testing the new flow on Jenkins by posting the new job spec up and telling it to pull Antrea from my branch. It seems that by pulling the same script out from Jenkins build has some environment complications introduced on Jenkins side. I'm still investigating this